### PR TITLE
feat: add upload file type 

### PR DESCRIPTION
**Pull request by AdrianBielecAriane**

### CODE REVIEW

File: packages/react-kit/src/components/form/Upload/BaseUpload.tsx [hunk 1]
```diff
@@ -276,11 +309,16 @@ function BaseUpload({
...
+      <FieldFileUploadWrapper
+        {...wrapperProps}
+        $isFileOnly={isFileOnly}
+        $disabled={!!disabled}
+      >
         <FieldInput
           {...props}
           hidden
           type="file"
+          id={name}
```
Comment (albertfolch-redeemeum) [inline packages/react-kit/src/components/form/Upload/BaseUpload.tsx:321]: we could have the same id with this approach, please change this or remove it

File: packages/react-kit/src/components/form/Upload/BaseUpload.tsx [hunk 2]
```diff
@@ -91,6 +93,9 @@ function BaseUpload({
   const isVideoOnly = mimetypes.every((mimetype) =>
     mimetype.startsWith("video/")
   );
+  const isFileOnly = mimetypes.every((mimetype) =>
+    mimetype.startsWith("application/")
```
Comment (albertfolch-redeemeum) [inline packages/react-kit/src/components/form/Upload/BaseUpload.tsx:97]:
```suggestion
    mimetype.startsWith("application/pdf")
```

File: packages/react-kit/src/components/form/Upload/BaseUpload.tsx [hunk 3]
```diff
@@ -91,6 +93,9 @@ function BaseUpload({
   const isVideoOnly = mimetypes.every((mimetype) =>
     mimetype.startsWith("video/")
   );
+  const isFileOnly = mimetypes.every((mimetype) =>
```
Comment (albertfolch-redeemeum) [inline packages/react-kit/src/components/form/Upload/BaseUpload.tsx:96]:
```suggestion
  const isPdfOnly = mimetypes.every((mimetype) =>
```

File: packages/react-kit/src/components/form/Upload/BaseUpload.tsx (hunk 2 above)
Comment (AdrianBielecAriane) [reply packages/react-kit/src/components/form/Upload/BaseUpload.tsx:97]: We do not care about docx / doc etc?

File: packages/react-kit/src/stories/buttons/Upload.stories.tsx [hunk 4]
```diff
@@ -88,3 +88,11 @@ export const VideoOnly = {
     withUpload: false
   } satisfies BaseUploadProps
 };
+
+export const FileOnly = {
```
Comment (albertfolch-redeemeum) [inline packages/react-kit/src/stories/buttons/Upload.stories.tsx:92]:
```suggestion
export const PdfOnly = {
```

File: packages/react-kit/src/components/form/Upload/BaseUpload.tsx (hunk 2 above)
Comment (albertfolch-redeemeum) [reply packages/react-kit/src/components/form/Upload/BaseUpload.tsx:97]: that's a question for Sukrita, but if you can then also upload a .js file and show the pdf logo it looks weird, if we only plan to use pdf then let's not overcomplicate I guess?

Comment (AdrianBielecAriane) [reply packages/react-kit/src/components/form/Upload/BaseUpload.tsx:97]: Totally agree with you, i change all. No time to think about it

Comment (albertfolch-redeemeum) [reply packages/react-kit/src/components/form/Upload/BaseUpload.tsx:97]: then let's make this change @AdrianBielecAriane

Comment (AdrianBielecAriane) [reply packages/react-kit/src/components/form/Upload/BaseUpload.tsx]: Done

**Review by albertfolch-redeemeum — COMMENTED** [review]
Id only show the Upload file button

image


and only show the file uploaded row once it's uploaded
image

check with @sukrita3 tho

File: packages/react-kit/src/components/form/Upload/BaseUpload.tsx [hunk 5]
```diff
@@ -347,6 +385,28 @@ function BaseUpload({
...
+                          style={{ display: "flex", justifyContent: "center" }}
+                          onClick={(e) => {
+                            e.preventDefault();
+                            e.stopPropagation();
+                            handleRemoveAllFiles();
+                          }}
+                        >
+                          <X size={12.5} />
+                        </button>
+                      </Grid>
```
Comment (albertfolch-redeemeum) [inline packages/react-kit/src/components/form/Upload/BaseUpload.tsx:409]: I think this should be themable/stylable, for example, what if in fermion we want this to have rounded edges but not in boson?

Comment (AdrianBielecAriane) [reply packages/react-kit/src/components/form/Upload/BaseUpload.tsx:409]: It is container for content only, `FileUploadWrapper` is still responsible for input look

**Comment by sukrita3** [comment]
Albert's suggestion makes sense, lets go with that.

**Review by albertfolch-redeemeum — COMMENTED** [review]
image

I think the file view was already implemented and now it looks broken, I chose 'multiple'

**Review by albertfolch-redeemeum — COMMENTED** [review]
also if I set disabled to true the cursor is still pointer

**Comment by albertfolch-redeemeum** [comment]
so if you upload one it looks like this
image

but if you upload 2 then
image


is this intended? it's weird the files look completely different @AdrianBielecAriane

**Comment by albertfolch-redeemeum** [comment]
regular upload button is broken now, the preview is not displayed
image

**Comment by AdrianBielecAriane** [comment]
> so if you upload one it looks like this image
> 
> but if you upload 2 then image
> 
> is this intended? it's weird the files look completely different @AdrianBielecAriane

It is intended, i do not know how should it look like i asked on slack Sukrita is it okay, she agreed on that so i thought it is ok.
I do not have figma so i cannot assume how it should like

**Comment by albertfolch-redeemeum** [comment]
> regular upload button is broken now, the preview is not displayed image


I think this is the video component with multiple set to true, let's leave this weird UI as it is out of scope and hence, ignore my comment

**Review by albertfolch-redeemeum — APPROVED** [review]

**Pull request merged**